www/app.py

```diff
@@ -75,70 +74,46 @@ class HeartBeatManager():
     :param payload:
         The json-formatted information about the rover and the user to send
         to rovercode-web.
-    :param session:
-        A requests object to store session info.
-    :param csrftoken:
-        The csrftoken provided by rovercode-web.
+    :param access_token:
+        The oauth access token being used.
     """
 
-    def __init__(self, payload, user_name, user_pass, id=None):
+    def __init__(self, client_id, client_secret, id=None):
         """Constructor for the HeartBeatManager."""
         self.run = True
         self.thread = None
         self.web_id = id
-        self.payload = payload
-        self.session = requests.session()
-
-        # Log in to rovercode-web
-        self.session.get(ROVERCODE_WEB_LOGIN_URL + '/')
-        try:
-            csrftoken = self.session.cookies['csrftoken']
-        except KeyError:
-            # if the server doesn't provide a csrftoken, no problem
-            csrftoken = ''
+        self.client_id = client_id
+        self.name = None
+        self.local_ip = get_local_ip()
 
+        # Log in to rovercode-web using oauth credentials
         login_data = {
-            'login':user_name,
-            'password':user_pass,
-            'csrfmiddlewaretoken':csrftoken,
+            'grant_type':'client_credentials',
+            'client_id':client_id,
+            'client_secret':client_secret,
         }
-        self.session.post(ROVERCODE_WEB_LOGIN_URL + '/', data=login_data)
-
-        # We have a new csrf token after logging in
-        try:
-            csrftoken = self.session.cookies['csrftoken']
-        except KeyError:
-            # if the server doesn't provide a csrftoken, no problem
-            csrftoken = ''
-        self.payload['csrfmiddlewaretoken'] = csrftoken
-        self.csrftoken = csrftoken
-
-    def create(self):
-        """Register the rover initially with rovercode-web."""
-        r = self.session.post(ROVERCODE_WEB_REG_URL + '/', self.payload)
-        info = json.loads(r.text)
-        try:
-            self.web_id = info['id']
-            init_input_gpio(info['left_eye_pin'], info['right_eye_pin'])
-            print "rovercode-web id is " + str(self.web_id)
-        except KeyError:
-            self.web_id = None
-        return r
+        r = requests.post(ROVERCODE_WEB_OAUTH2_URL + '/', data=login_data)
+        print r.text
```

Comment:
Do you want either of these print statements?

Reply:
Nope! Removed.